File: back/back/Models/Specialite.cs
using LiteDB;

namespace back.Models;

public sealed class Specialite
{
    [BsonId]
    public int Id { get; set; }

    [BsonRef]
    public List<Specialite> ListeParent { get; set; } = [];

    [BsonRef]
    public Grade Grade { get; set; } = null!;
    public string Nom { get; set; } = null!;
    public string Raccourci { get; set; } = null!;
    public string Categorie { get; set; } = null!;
    public string? Description { get; set; }
    public bool EstNavy { get; set; }
    public string? NomImage { get; set; }
}

File: front/src/models/Specialite.ts
export type Specialite =
{
    id: number,
    idParents: number[],
    nom: string,
    raccourci: string,
    categorie: string,
    grade: {
        id: number,
        nom: string,
        nbOperationRequis: number
    }
    description: string,
    estNavy: boolean,
    urlImage: string
}

export type SpecialiteLeger = 
{
    id: number,
    nom: string
}

export type SpecialiteRequete =
{
    idParents: number[],
    nom: string,
    raccourci: string,
    categorie?: string,
    idGrade: number,
    description: string
}